src/components/molecules/button/button.ts

```diff
@@ -0,0 +1,103 @@
+import { LitElement, html } from 'lit'
+import { customElement, property } from 'lit/decorators.js'
+
+import type { MsIcons } from '~/assets/icons'
+import { Icons } from '~/assets/icons'
+import { WithTheme } from '~/mixins'
+import { getThemeColor } from '~/utils'
+
+import { msButtonstyles } from './button.styles'
+
+@customElement('ms-button')
+export class MSButton extends WithTheme(LitElement) {
+  static styles = msButtonstyles
+
+  @property({ type: String })
+  variant = 'contained'
+
+  @property({ type: String })
+  color =
+    this.variant === 'contained'
+      ? getThemeColor('gray-white').toString()
+      : getThemeColor('regular-user-primary-main').toString()
```

Comment:
are those `.toString()` really needed? I guess `getThemeColor` returns something link 'var(--color,red)', if not, ignore my comment